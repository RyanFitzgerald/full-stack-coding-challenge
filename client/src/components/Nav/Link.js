import styled from 'styled-components';
import { Link as RouterLink } from 'react-router-dom';

const Link = styled(RouterLink)`
  border-radius: 4px;
  color: #696d80;
  display: inline-block;
  font-weight: 700;
  padding: 15px;
  text-decoration: none;
  transition: 0.25s ease all;

  &:hover {
    background: #f2f2f2;
    color: #3498db;
  }
`;

export default Link;