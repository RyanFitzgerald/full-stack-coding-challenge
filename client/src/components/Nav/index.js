import React from 'react';

// Import components
import Link from './Link';

function Nav() {
  return (
    <nav>
      <Link to="/">Search</Link>
      <Link to="/add">Add</Link>
    </nav>
  );
}

export default Nav;