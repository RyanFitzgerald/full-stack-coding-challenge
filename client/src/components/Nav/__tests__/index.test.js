import React from 'react';
import { StaticRouter } from 'react-router-dom';

import Nav from '../index';

describe('<Nav />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <StaticRouter location="/">
        <Nav/>
      </StaticRouter>
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should contain 2 links', () => {
    const renderedComponent = mount(
      <StaticRouter location="/">
        <Nav/>
      </StaticRouter>
    );
    
    expect(renderedComponent.find('Link')).toHaveLength(2);
  });
});