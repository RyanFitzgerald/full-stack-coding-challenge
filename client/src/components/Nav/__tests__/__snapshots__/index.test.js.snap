// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should match the snapshot 1`] = `
.c0 {
  border-radius: 4px;
  color: #696d80;
  display: inline-block;
  font-weight: 700;
  padding: 15px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.25s ease all;
  transition: 0.25s ease all;
}

.c0:hover {
  background: #f2f2f2;
  color: #3498db;
}

<nav>
  <a
    className="c0"
    href="/"
    onClick={[Function]}
  >
    Search
  </a>
  <a
    className="c0"
    href="/add"
    onClick={[Function]}
  >
    Add
  </a>
</nav>
`;
